project-base/tests/App/Functional/Twig/FilesystemLoaderTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\App\Functional\Twig;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Setting\Setting;
+use Tests\App\Test\FunctionalTestCase;
+
+class FilesystemLoaderTest extends FunctionalTestCase
+{
+    public function testLoadingMultiDesignTemplate()
+    {
+        $twigFilesystemLoader = $this->getContainer()->get('twig.loader.native_filesystem');
+        $setting = $this->getContainer()->get(Setting::class);
+        $twig = $this->getContainer()->get('twig');
+
+        $domainConfig = clone $this->domain->getCurrentDomainConfig();
+        $domainConfigReflection = new \ReflectionClass($domainConfig);
+        $designIdReflection = $domainConfigReflection->getProperty('designId');
+        $designIdReflection->setAccessible(true);
+        $designIdReflection->setValue($domainConfig, 'custom');
+
+        $domain = new Domain([$domainConfig], $setting);
+        $domain->switchDomainById(1);
+
+        $twigReflection = new \ReflectionClass($twigFilesystemLoader);
+        $domainReflection = $twigReflection->getProperty('domain');
+        $domainReflection->setAccessible(true);
+        $domainReflection->setValue($twigFilesystemLoader, $domain);
+
+        $fileOutput = $twig->render('Front/Tests/twigLoader.html.twig');
+
+        $domainReflection->setValue($twigFilesystemLoader, $this->domain);
+        $domainReflection->setAccessible(false);
+
+        $this->assertSame('Hello world', trim($fileOutput));
```

Comment:
Thanks for the test! 

What do you think of the following?

```php
$setting = $this->createMock(Setting::class);
        $twig = $this->getContainer()->get('twig');

        $twigFilesystemLoader = $this->getContainer()->get('twig.loader.native_filesystem');
        $twigFilesystemLoader->addPath(__DIR__ . '/Resources');

        $domainConfigWithoutCustom = new DomainConfig(
            1,
            'http://webserver:8080',
        'domain1',
            'en',
            'sa',
            ''
        );

        $domainConfigWithCustom = new DomainConfig(
            2,
            'http://webserver:8080',
            'domain2',
            'en',
            'sa',
            'custom'
        );

        $domain = new Domain([$domainConfigWithoutCustom, $domainConfigWithCustom], $setting);

        $twigReflection = new \ReflectionClass($twigFilesystemLoader);
        $domainReflection = $twigReflection->getProperty('domain');
        $domainReflection->setAccessible(true);
        $domainReflection->setValue($twigFilesystemLoader, $domain);

        $domain->switchDomainById(1);
        $fileOutput = $twig->render('Front/twigLoader.html.twig');
        self::assertSame('Hello world', trim($fileOutput));

        $domain->switchDomainById(2);
        $fileOutput = $twig->render('Front/twigLoader.html.twig');
        self::assertSame('Hello world custom', trim($fileOutput));
```

It has several advantages. 
 - I don't need settings service registered as public (used mock)
 - I test custom designId and normal (no designId)
 - Template used for tests only are in test folder (project-base/tests/App/Functional/Twig/Resources/Front/twigLoader.custom.html.twig and project-base/tests/App/Functional/Twig/Resources/Front/twigLoader.html.twig have to be created)
 - less reflection for domainconfig as it's created manually

Reply:
Ok I have updated it